test/e2e/actions/actions.go

```diff
@@ -4,87 +4,66 @@ package actions
 
 import (
 	"fmt"
-	"strconv"
 	"time"
 
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/actions/kube"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+
 	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/api/atlas"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-	. "github.com/onsi/gomega/gbytes"
 
-	appclient "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/appclient"
-	kubecli "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/cli/kubecli"
 	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/model"
-	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/utils"
 )
 
-func UpdateDeployment(newData *model.TestDataProvider) {
-	var generation int
-	By("Update Deployment\n", func() {
-		utils.SaveToFile(
-			newData.Resources.Deployments[0].DeploymentFileName(newData.Resources),
-			utils.JSONToYAMLConvert(newData.Resources.Deployments[0]),
-		)
-		generation, _ = strconv.Atoi(kubecli.GetGeneration(newData.Resources.Namespace, newData.Resources.Deployments[0].GetDeploymentNameResource()))
-		kubecli.Apply(newData.Resources.Deployments[0].DeploymentFileName(newData.Resources), "-n", newData.Resources.Namespace)
-		generation++
-	})
-
-	By("Wait Deployment updating\n", func() {
-		WaitDeployment(newData.Resources, strconv.Itoa(generation))
-	})
-
-	By("Check attributes\n", func() {
-		aClient := atlas.GetClientOrFail()
-		uDeployment := aClient.GetDeployment(newData.Resources.ProjectID, newData.Resources.Deployments[0].Spec.GetDeploymentName())
-		CompareDeploymentsSpec(newData.Resources.Deployments[0].Spec, uDeployment)
-	})
-}
-
-func UpdateDeploymentFromUpdateConfig(data *model.TestDataProvider) {
-	By("Load new Deployment config", func() {
-		data.Resources.Deployments = []model.AtlasDeployment{} // TODO for range
-		GinkgoWriter.Write([]byte(data.ConfUpdatePaths[0]))
-		data.Resources.Deployments = append(data.Resources.Deployments, model.LoadUserDeploymentConfig(data.ConfUpdatePaths[0]))
-		data.Resources.Deployments[0].Spec.Project.Name = data.Resources.Project.GetK8sMetaName()
-		utils.SaveToFile(
-			data.Resources.Deployments[0].DeploymentFileName(data.Resources),
-			utils.JSONToYAMLConvert(data.Resources.Deployments[0]),
-		)
-	})
-
-	UpdateDeployment(data)
-
-	By("Check user data still in the Deployment\n", func() {
-		for i := range data.Resources.Users { // TODO in parallel(?)
-			port := strconv.Itoa(i + data.PortGroup)
-			key := port
-			data := fmt.Sprintf("{\"key\":\"%s\",\"shipmodel\":\"heavy\",\"hp\":150}", key)
-			app := appclient.NewTestAppClient(port)
-			Expect(app.Get("/mongo/" + key)).Should(Equal(data))
+func UpdateDeploymentSpecByIndex(spec v1.AtlasDeploymentSpec, index int) func(data *model.TestDataProvider) {
```

Comment:
Have to say this isn't much better than the initial version as both function name and definition do not explain why there is an `index` in the `spec` and what does it mean 🙃

Reply:
Renamed function. May you look at it again?

Reply:
Well I wanted to change it completely and separate this logic into two separate functions, where one selects the deployment from initial deployments and the other one update the deployment which you give it (pre-selected) but at this stage I think it looks fine as is 👌